nextcord/http.py

```diff
@@ -204,33 +653,56 @@ async def ws_connect(self, url: str, *, compress: int = 0) -> Any:
 
         return await self.__session.ws_connect(url, **kwargs)
 
+    # ruff: noqa: PLR0915
+    # Ruff says this is too many statements in a single function, but I'm not sure how to break it down while
```

Comment:
You could remove the parts of the code that make the actual HTTP request & handle the HTTP errors into an internal function then have this request function just be dedicated to handling ratelimits.

Reply:
ok.